spec/models/resource_action_serializer_spec.rb

```diff
@@ -2,33 +2,22 @@
   let(:resource_action_serializer) { described_class.new }
 
   describe "#serialize" do
-    let(:resource_action) do
-      ResourceAction.new(
-        :dialog_id     => 123,
-        :resource_id   => 321,
-        :created_at    => Time.now,
-        :updated_at    => Time.now,
-        :resource_type => "DialogField",
-        :ae_namespace  => "Customer/Sample",
-        :ae_class      => "Methods",
-        :ae_instance   => "Testing"
-      )
-    end
+    let(:resource_action) { ResourceAction.new(expected_serialized_values) }
 
     let(:expected_serialized_values) do
       {
-        "action"        => nil,
         "resource_type" => "DialogField",
         "ae_namespace"  => "Customer/Sample",
         "ae_class"      => "Methods",
         "ae_instance"   => "Testing",
-        "ae_message"    => nil,
-        "ae_attributes" => {}
       }
     end
 
     it "serializes the resource_action" do
-      expect(resource_action_serializer.serialize(resource_action)).to eq(expected_serialized_values)
+      serialized = resource_action_serializer.serialize(resource_action)
+      expect(serialized).to have_attributes(expected_serialized_values)
+      expect(serialized.keys).to include("action", "ae_attributes", "ae_message")
+      expect(serialized.keys).not_to include(*described_class::EXCLUDED_ATTRIBUTES)
```

Comment:
Much nicer, and it happens to be more specific in what the expected results are!